environment/src/helpers/resources.rs

```diff
@@ -190,6 +202,33 @@ impl Resources {
         // This channel will not be closed before the final shutdown.
         let _ = self.sender.send(ResourceRequest::LogSummary);
     }
+
+    /// Instruments a task with the given `handle` and `name`; logs useful metrics every `interval`.
+    #[cfg(feature = "task-metrics")]
+    fn instrument_task(&self, handle: tokio::task::JoinHandle<()>, name: &'static str, interval: Duration) {
+        let task_monitor = tokio_metrics::TaskMonitor::new();
+
+        let tm = task_monitor.clone();
+        let logging_task = tokio::spawn(async move {
+            loop {
+                tokio::time::sleep(interval).await;
+                let metrics = tm.cumulative();
+                debug!(
```

Comment:
Might be worth mentioning this list isn't exhaustive? https://docs.rs/tokio-metrics/latest/tokio_metrics/struct.TaskMetrics.html#

Reply:
Sure, why not; I'll add a comment when I rebase this PR once https://github.com/AleoHQ/snarkOS/pull/1652 is merged.